src/test/components/TimelineMarkers.test.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React from 'react';
+import TimelineMarkers from '../../content/containers/TimelineMarkers';
+import renderer from 'react-test-renderer';
+import { Provider } from 'react-redux';
+import mockCanvasContext from '../fixtures/mocks/canvas-context';
+import { storeWithProfile } from '../fixtures/stores';
+import { getProfileWithMarkers } from '../store/fixtures/profiles';
+
+jest.useFakeTimers();
+
+it('renders TimelineMarkers correctly', () => {
+  window.requestAnimationFrame = fn => setTimeout(fn, 0);
+  window.devicePixelRatio = 1;
+  const ctx = mockCanvasContext();
+
+  /**
+   * Mock out any created refs for the components with relevant information.
+   */
+  function createNodeMock(element) {
+    // <TimelineCanvas><canvas /></TimelineCanvas>
+    if (element.type === 'canvas') {
+      return {
+        getBoundingClientRect: () => _getBoundingBox(200, 300),
+        getContext: () => ctx,
+        style: {},
+      };
+    }
+    // <TimelineViewport />
+    if (element.props.className.split(' ').includes('timelineViewport')) {
+      return {
+        getBoundingClientRect: () => _getBoundingBox(200, 300),
+      };
+    }
+    return null;
+  }
+
+  const profile = getProfileWithMarkers([
+    ['Marker A', 0, {startTime: 0, endTime: 10}],
+    ['Marker B', 0, {startTime: 0, endTime: 10}],
+    ['Marker C', 0, {startTime: 5, endTime: 15}],
+  ]);
+
+  const timeline = renderer.create(
+    <Provider store={storeWithProfile(profile)}>
+      <TimelineMarkers threadIndex={0} viewHeight={1000} />
+    </Provider>,
+    {createNodeMock}
+  );
+
+  jest.runAllTimers();
+  
+  const tree = timeline.toJSON();
+  const drawCalls = ctx.__flushDrawLog();
+
+  expect(tree).toMatchSnapshot();
+  expect(drawCalls).toMatchSnapshot();
+
+  delete window.requestAnimationFrame;
+  delete window.devicePixelRatio;
```

Comment:
pretty sure this will bite us later :/ I'll have a look at how to make this better